tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,544 @@
+#include <app_keys.h>
+
+#include <zephyr/logging/log.h>
+#define LOG_MODULE_NAME bttester_vcp
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+#include "btp/btp.h"
+
+#define CONTROLLER_INDEX 0
+
+struct bt_vcp_vol_rend_register_param vcp_register_param;
+struct bt_vcp_included included;
+static bool g_aics_active = 1;
+
+static void set_register_params(uint8_t gain_mode);
+
+/* Volume Control Service */
+static void vcs_supported_commands(uint8_t *data, uint16_t len)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(BTP_DATA_MAX_SIZE);
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, VCS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, VCS_INIT);
+	net_buf_simple_add_u8(buf, VCS_SET_VOL);
+	net_buf_simple_add_u8(buf, VCS_VOL_UP);
+	net_buf_simple_add_u8(buf, VCS_VOL_DOWN);
+	net_buf_simple_add_u8(buf, VCS_MUTE);
+	net_buf_simple_add_u8(buf, VCS_UNMUTE);
+
+	tester_send(BTP_SERVICE_ID_VCS, VCS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, buf->data, buf->len);
+}
+
+static void set_volume(uint8_t *data, uint16_t len)
+{
+	const struct vcs_set_vol_cmd *cmd = (void *) data;
+	uint8_t status = BTP_STATUS_SUCCESS;
+	uint32_t volume;
+	int err;
+
+	volume = sys_le32_to_cpu(cmd->volume);
+
+	LOG_DBG("volume 0x%04x", volume);
+
+	err = bt_vcp_vol_rend_set_vol(volume);
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_SET_VOL, CONTROLLER_INDEX, status);
+}
+
+static void vol_up(void)
+{
+	uint8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	LOG_DBG("Volume Up");
+
+	err = bt_vcp_vol_rend_vol_up();
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_UP, CONTROLLER_INDEX, status);
+}
+
+static void vol_down(void)
+{
+	uint8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	LOG_DBG("Volume Down");
+
+	err = bt_vcp_vol_rend_vol_down();
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_DOWN, CONTROLLER_INDEX, status);
+}
+
+static void mute(void)
+{
+	uint8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	err = bt_vcp_vol_rend_mute();
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_MUTE, CONTROLLER_INDEX, status);
+}
+
+static void unmute(void)
+{
+	uint8_t status = BTP_STATUS_SUCCESS;
+	int err;
+
+	err = bt_vcp_vol_rend_unmute();
+	if (err) {
+		status = BTP_STATUS_FAILED;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_UNMUTE, CONTROLLER_INDEX, status);
+}
+
+static void vcs_state_cb(int err, uint8_t volume, uint8_t mute)
+{
+	if (err) {
+		LOG_DBG("VCP state cb err (%d)", err);
+		return;
+	}
+}
+
+static void vcs_flags_cb(int err, uint8_t flags)
+{
+	if (err) {
+		LOG_DBG("VCP flags cb err (%d)", err);
+		return;
+	}
+}
+
+static struct bt_vcp_vol_rend_cb vcs_cb = {
+	.state = vcs_state_cb,
+	.flags = vcs_flags_cb,
+};
+
+static void vcp_init(void)
+{
+	struct bt_vcp_vol_rend_register_param vcp_register_param;
+	int err;
+
+	/* Default values */
+	vcp_register_param.step = 1;
```

Comment:
Is the step size an IXIT value?

Reply:
The C files here don't read from any files or anything? How are we keeping the C source files and the IXIT values in sync?

Reply:
I think (but I'm not sure I'll have to confirm this) that the way to do this might be in a function that initializes the service, there we can pass an argument from auto-pts regarding step size. How does that sound?

Reply:
> The C files here don't read from any files or anything? How are we keeping the C source files and the IXIT values in sync?

That's a good question. The IXIT values are defined in python script like here: https://github.com/intel/auto-pts/blob/8e2d7eec9e7125b09f67b08704aa591c96dd4004/autopts/ptsprojects/zephyr/l2cap.py#L45. Those are set to meet the configuration of IUT.
If test needs to be ran with different settings than default, we just update the IXIT. @szymon-czapracki Is there a test case that requires usage of values different than default?

Maybe we could have a commands to set/get the configuration?

Reply:
What I've done earlier in a similar setup, was to have to IXIT values in a config file, which were read by both the C code and the "autoPTS" code, so the values were only defined a single place. Given that this is in multiple projects, this may be harder to do for Zephyr. 

> Maybe we could have a commands to set/get the configuration?

That could be option, assuming that we are not using any of the IXIT values as compile values, i.e. values that cannot be modified at runtime.